guide/installation-and-configuration/upgrading.markdown

```diff
@@ -5,42 +5,67 @@ published: true
 sorting: 30
 ---
 
+This guide documents our recommendation on how to upgrade an existing
+installation of CFEngine Enterprise to {{site.cfengine.branch}}. Community users
+can use these instructions as a guide skipping the parts that are not relevant.
 
-# Summary of the upgrade process
+In short, the steps are:
 
-In short, the steps are the following:
+1. [Backup][Upgrading#Backup]
+2. [Masterfiles Policy Framework upgrade][Upgrading#Masterfile Policy Framework upgrade]
+3. [Enterprise Hub binary upgrade][Upgrading#Enterprise Hub binary upgrade]
+4. [Agent binary upgrade][Upgrading#Agent binary upgrade]
 
-1. Before upgrading any package, **upgrade your masterfiles** on the hub
-   (suggested download is the "Masterfiles ready-to-install tarball")
-   * Make sure you port whatever changes you had in the old
-     masterfiles, to the new masterfiles version
-   * Test that the new policy works properly before putting it in
-     `/var/cfengine/masterfiles` directory and deploying it
-2. Wait until the new masterfiles have propagated to all clients,
-   and ensure there are no errors
-3. Upgrade the CFEngine package on the hub with the new version
-4. Upgrade the CFEngine package on all clients
+**Notes:**
 
-**NOTE**: This is also the recommended way for upgrades between **minor
-releases** for example 3.10.0 to 3.10.2.
+- Upgrades are supported from any [currently supported version][supported versions].
 
-**NOTE**: Upgrading between major LTS releases is safest to be done
-**step-by-step from one LTS version to the next**, for example from
-3.6.x to 3.7.x to 3.10.x. Try not to do multi-version upgrade at once.
+## Backup
 
+Take a full backup from your policy hub before making any changes so that you
+can recover if anything goes wrong.
 
-# Detailed upgrade process
+1. Stop the CFEengine services.
 
-This guide documents our recommendation on how to upgrade an existing
-installation of CFEngine Enterprise 3.7.x to {{site.cfengine.branch}}. Community
-users can use these instructions as a guide skipping the parts that are not
-relavent.
+   For systemd managed systems:
+
+   ```console
+   root@hub:~# systemctl cfengine3 stop
+   ```
+
+   For SysVinit:
+
+    ```console
+   root@hub:~# service stop cfengine3
+   ```
+
+2. Create an archive containing all cfengine information.
+
+   Ensure you have enough disk space where your backup archive will be created.
+
+   ```console
+   root@hub:~# tar -czf /tmp/$(date +%Y-%m-%d)-cfengine-full-backup.tar.gz /var/cfengine /opt/cfengine
```

Comment:
I'd suggest `/var/tmp` here even though it's copied over somewhere else later. Somebody may have the great idea to reboot the system as part of the upgrade and `/tmp` is quite often *tmpfs* these days.